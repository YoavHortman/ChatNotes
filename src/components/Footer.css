.Footer__root {
    flex-direction: column;
    display: flex;
    flex: 0 0 auto;
}

.Footer__input {
    border-radius: 10px;
    border: 2px solid black;
    padding: 10px;
    margin: 10px;
    font-size: 18px;
}

.Footer_err {
    color: red;
    font-size: 18px;
    font-weight: bold;
    text-align: center;
}