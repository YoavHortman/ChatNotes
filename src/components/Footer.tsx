import * as React from "react";
import "./Footer.css"


interface FooterProps {
    onCreateMessage: (messageText: string) => void;
}

interface FooterState {
    text: string | undefined;
    errMessage: string | null;
}

export class Footer extends React.Component<FooterProps, FooterState> {
    constructor(props: FooterProps) {
        super(props);
        this.state = {
            text: undefined,
            errMessage: null
        }
    }

    handleTextChange(newVal: string) {
        this.setState({text: newVal, errMessage: null});
    }

    handleKeyPress(key: string) {
        if (key === "Enter") {
            if (this.state.text !== undefined && this.state.text.length > 0) {
                this.props.onCreateMessage(this.state.text);
                this.setState({text: undefined})
            } else {
                console.log("no input");
                this.setState({errMessage: "No input.."})
            }
        }
    }


    render() {
        return (
            <div className={"Footer__root"}>
                <input
                    type={"text"}
                    value={this.state.text}
                    onChange={(e) => this.handleTextChange(e.target.value)}
                    onKeyPress={(e) => this.handleKeyPress(e.key)}
                    placeholder={"Enter your message..."}
                    className={"Footer__input"}
                />
                {this.state.errMessage === null ? null :
                    <div className={"Footer_err"}>
                        {this.state.errMessage}
                    </div>
                }

            </div>
        );
    }
}
