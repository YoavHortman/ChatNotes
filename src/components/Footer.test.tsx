import * as React from "react";
import * as renderer from "react-test-renderer";
import {Footer} from "./Footer";

it('Should render', () => {
    const tree = renderer
        .create(<Footer onCreateMessage={() => {
        }}/>).toJSON();

    expect(tree).toMatchSnapshot();
});