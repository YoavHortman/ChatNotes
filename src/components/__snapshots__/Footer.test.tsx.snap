// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render 1`] = `
<div
  className="Footer__root"
>
  <input
    className="Footer__input"
    onChange={[Function]}
    onKeyPress={[Function]}
    placeholder="Enter your message..."
    type="text"
    value=""
  />
</div>
`;
